### ISSUE TITLE
[Exception] IndexNotReadyException

### ISSUE THREAD

**Issue by artspb** [Haehnchen/idea-php-laravel-plugin#109]
It seems you invoke resolve when indices are built. I have a couple of those.

```
Error while indexing D:\Code\LocalReachPros\Dashboard\v7\routes\web.php
To reindex this file IDEA has to be restarted

com.intellij.openapi.project.IndexNotReadyException: Please change caller according to com.intellij.openapi.project.IndexNotReadyException documentation
	at com.intellij.util.indexing.FileBasedIndexImpl.handleDumbMode(FileBasedIndexImpl.java:720)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:669)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:652)
	at com.intellij.psi.stubs.StubIndexImpl.doProcessStubs(StubIndexImpl.java:346)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:327)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:146)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:135)
	at com.jetbrains.php.PhpIndexImpl.getByName(PhpIndexImpl.java:571)
	at com.jetbrains.php.PhpIndexImpl.getByFQN(PhpIndexImpl.java:551)
	at com.jetbrains.php.PhpIndexImpl.getClassesByFQN(PhpIndexImpl.java:581)
	at com.jetbrains.php.PhpIndexImpl.getAnyByFQN(PhpIndexImpl.java:597)
	at com.jetbrains.php.PhpIndexImpl.getBySignatureInternal(PhpIndexImpl.java:424)
	at com.jetbrains.php.PhpIndexImpl.getBySignature(PhpIndexImpl.java:319)
	at com.jetbrains.php.PhpIndexImpl.getClasses(PhpIndexImpl.java:507)
	at com.jetbrains.php.PhpIndexImpl.getBySignatureInternal(PhpIndexImpl.java:443)
	at com.jetbrains.php.PhpIndexImpl.getBySignature(PhpIndexImpl.java:319)
	at com.jetbrains.php.PhpIndexImpl.getClasses(PhpIndexImpl.java:507)
	at com.jetbrains.php.PhpIndexImpl.getBySignatureInternal(PhpIndexImpl.java:443)
	at com.jetbrains.php.PhpIndexImpl.getBySignature(PhpIndexImpl.java:319)
	at com.jetbrains.php.PhpIndexImpl.completeType(PhpIndexImpl.java:79)
	at com.jetbrains.php.lang.psi.resolve.types.PhpType.globalLocationAware(PhpType.java:413)
	at com.jetbrains.php.lang.psi.elements.impl.MemberReferenceImpl.resolveGlobal(MemberReferenceImpl.java:147)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl$1.resolve(PhpReferenceImpl.java:144)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl$1.resolve(PhpReferenceImpl.java:141)
	at com.intellij.psi.impl.source.resolve.ResolveCache$3.compute(ResolveCache.java:142)
	at com.intellij.openapi.util.RecursionManager$2.doPreventingRecursion(RecursionManager.java:112)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:139)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:167)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:159)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl.multiResolve(PhpReferenceImpl.java:136)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl.resolve(PhpReferenceImpl.java:122)
	at fr.adrienbrault.idea.symfony2plugin.Symfony2InterfacesUtil.getMultiResolvedMethod(Symfony2InterfacesUtil.java:166)
	at fr.adrienbrault.idea.symfony2plugin.util.MethodMatcher$StringParameterRecursiveMatcher.match(MethodMatcher.java:154)
	at fr.adrienbrault.idea.symfony2plugin.util.MethodMatcher.getMatchedSignatureWithDepth(MethodMatcher.java:42)
	at de.espend.idea.laravel.routing.utils.RouteGroupUtil.lambda$getRouteGroupPropertiesCollection$0(RouteGroupUtil.java:37)
	at com.intellij.psi.util.PsiTreeUtil.findFirstParent(PsiTreeUtil.java:330)
	at de.espend.idea.laravel.routing.utils.RouteGroupUtil.getRouteGroupPropertiesCollection(RouteGroupUtil.java:36)
	at de.espend.idea.laravel.routing.utils.RoutingUtil$RouteNamePsiRecursiveElementVisitor.getRouteNamePrefix(RoutingUtil.java:152)
	at de.espend.idea.laravel.routing.utils.RoutingUtil$RouteNamePsiRecursiveElementVisitor.visitElement(RoutingUtil.java:137)
	at com.jetbrains.php.lang.psi.elements.impl.MethodReferenceImpl.accept(MethodReferenceImpl.java:32)
	at com.intellij.psi.impl.PsiElementBase.acceptChildren(PsiElementBase.java:69)
	at com.intellij.psi.PsiRecursiveElementVisitor.visitElement(PsiRecursiveElementVisitor.java:40)
	at de.espend.idea.laravel.routing.utils.RoutingUtil$RouteNamePsiRecursiveElementVisitor.visitElement(RoutingUtil.java:143)
	at com.jetbrains.php.lang.psi.elements.impl.StatementImpl.accept(StatementImpl.java:24)
	at com.intellij.psi.impl.source.tree.LazyParseablePsiElement.acceptChildren(LazyParseablePsiElement.java:106)
	at com.intellij.psi.PsiRecursiveElementVisitor.visitElement(PsiRecursiveElementVisitor.java:40)
	at de.espend.idea.laravel.routing.utils.RoutingUtil$RouteNamePsiRecursiveElementVisitor.visitElement(RoutingUtil.java:143)
	at com.jetbrains.php.lang.psi.elements.impl.GroupStatementImpl.accept(GroupStatementImpl.java:30)
	at com.intellij.psi.impl.PsiElementBase.acceptChildren(PsiElementBase.java:69)
	at com.intellij.psi.PsiRecursiveElementVisitor.visitElement(PsiRecursiveElementVisitor.java:40)
	at de.espend.idea.laravel.routing.utils.RoutingUtil$RouteNamePsiRecursiveElementVisitor.visitElement(RoutingUtil.java:143)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
	at com.jetbrains.php.lang.psi.elements.impl.PhpPsiElementImpl.accept(PhpPsiElementImpl.java:73)
	at com.jetbrains.php.lang.psi.elements.impl.FunctionImpl.accept(FunctionImpl.java:118)
	at com.intellij.psi.impl.PsiElementBase.acceptChildren(PsiElementBase.java:69)
	at com.intellij.psi.PsiRecursiveElementVisitor.visitElement(PsiRecursiveElementVisitor.java:40)
	at de.espend.idea.laravel.routing.utils.RoutingUtil$RouteNamePsiRecursiveElementVisitor.visitElement(RoutingUtil.java:143)
	at com.jetbrains.php.lang.psi.elements.impl.PhpExpressionImpl.accept(PhpExpressionImpl.java:43)
	at com.intellij.psi.impl.PsiElementBase.acceptChildren(PsiElementBase.java:69)
	at com.intellij.psi.PsiRecursiveElementVisitor.visitElement(PsiRecursiveElementVisitor.java:40)
	at de.espend.idea.laravel.routing.utils.RoutingUtil$RouteNamePsiRecursiveElementVisitor.visitElement(RoutingUtil.java:143)
	at com.jetbrains.php.lang.psi.elements.impl.ParameterListImpl.accept(ParameterListImpl.java:33)
	at com.intellij.psi.impl.PsiElementBase.acceptChildren(PsiElementBase.java:69)
	at com.intellij.psi.PsiRecursiveElementVisitor.visitElement(PsiRecursiveElementVisitor.java:40)
	at de.espend.idea.laravel.routing.utils.RoutingUtil$RouteNamePsiRecursiveElementVisitor.visitElement(RoutingUtil.java:143)
	at com.jetbrains.php.lang.psi.elements.impl.MethodReferenceImpl.accept(MethodReferenceImpl.java:32)
	at com.intellij.psi.impl.PsiElementBase.acceptChildren(PsiElementBase.java:69)
	at com.intellij.psi.PsiRecursiveElementVisitor.visitElement(PsiRecursiveElementVisitor.java:40)
	at de.espend.idea.laravel.routing.utils.RoutingUtil$RouteNamePsiRecursiveElementVisitor.visitElement(RoutingUtil.java:143)
	at com.jetbrains.php.lang.psi.elements.impl.StatementImpl.accept(StatementImpl.java:24)
	at com.intellij.psi.impl.PsiElementBase.acceptChildren(PsiElementBase.java:69)
	at com.intellij.psi.PsiRecursiveElementVisitor.visitElement(PsiRecursiveElementVisitor.java:40)
	at de.espend.idea.laravel.routing.utils.RoutingUtil$RouteNamePsiRecursiveElementVisitor.visitElement(RoutingUtil.java:143)
	at com.jetbrains.php.lang.psi.elements.impl.GroupStatementSimpleImpl.accept(GroupStatementSimpleImpl.java:28)
	at com.intellij.psi.impl.source.tree.SharedImplUtil.acceptChildren(SharedImplUtil.java:200)
	at com.intellij.psi.impl.source.PsiFileImpl.acceptChildren(PsiFileImpl.java:878)
	at de.espend.idea.laravel.routing.utils.RoutingUtil.visitRoutesForAs(RoutingUtil.java:100)
	at de.espend.idea.laravel.routing.utils.RoutingUtil.getRoutesAsNames(RoutingUtil.java:92)
	at de.espend.idea.laravel.stub.RouteIndexExtension.lambda$getIndexer$0(RouteIndexExtension.java:42)
	at com.intellij.util.indexing.impl.MapReduceIndex.mapInput(MapReduceIndex.java:251)
	at com.intellij.util.indexing.VfsAwareMapReduceIndex.calculateUpdateData(VfsAwareMapReduceIndex.java:96)
	at com.intellij.util.indexing.impl.MapReduceIndex.update(MapReduceIndex.java:194)
	at com.intellij.util.indexing.FileBasedIndexImpl.updateSingleIndex(FileBasedIndexImpl.java:1587)
	at com.intellij.util.indexing.FileBasedIndexImpl.lambda$doIndexFileContent$17(FileBasedIndexImpl.java:1523)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.freezeFileTypeTemporarilyIn(FileTypeManagerImpl.java:516)
	at com.intellij.util.indexing.FileBasedIndexImpl.doIndexFileContent(FileBasedIndexImpl.java:1484)
	at com.intellij.util.indexing.FileBasedIndexImpl.indexFileContent(FileBasedIndexImpl.java:1469)
	at com.intellij.util.indexing.UnindexedFilesUpdater.lambda$indexFiles$0(UnindexedFilesUpdater.java:105)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.lambda$run$0(CacheUpdateRunner.java:229)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1131)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.lambda$run$1(CacheUpdateRunner.java:246)
	at com.intellij.openapi.progress.impl.CoreProgressManager$3.run(CoreProgressManager.java:181)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:586)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:531)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:166)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.run(CacheUpdateRunner.java:242)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:332)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by Kurre** [comment]
Probably related to #108

**Comment by adelf** [comment]
Yes. Thanks. Working on it right now :)

**Comment by Haehnchen** [comment]
fixed by @adelf in `0.13.2`
